### ISSUE TITLE
[SECURITY] Cross-Site Scripting (XSS) Vulnerability in Template

### ISSUE THREAD

**Issue by PKUcoldkeyboard** [colinwilson/lotusdocs#70]
### Past Issues Searched

- [X] I have searched open and closed issues to make sure that the bug has not yet been reported

### Issue is a Bug Report

- [X] This is a bug report and not a feature request, nor asking for support

### Describe the bug

Hello,

I have identified a cross-site scripting (XSS) vulnerability in your Hugo template. Attackers can exploit this vulnerability by injecting code similar to `<a href="javascript:alert('XSS')">Click me</a>` into the search box.

Vulnerability Description:

Vulnerability Type: Cross-Site Scripting (XSS)
Affected Area: Search box
Attack Scenario: Attackers can insert malicious code into the search box, which gets executed when users visit the search results page.

Please address this issue as soon as possible to enhance the security of your template. Thank you for your attention！

### Expected behavior

When a user inputs any code or script into the search box, it should be treated as plain text and displayed as such in the search results page. The code or script should not be executed or interpreted as HTML or JavaScript. This ensures that the search functionality remains secure and prevents any potential cross-site scripting (XSS) attacks.

### Environment

```markdown
- OS: Win11
- Browser: Google
- Browser Version: 115
```

**Comment by PKUcoldkeyboard** [comment]
Sorry,  I saw the unsafe setting. Thank you for your template. Very good

**Comment by PKUcoldkeyboard** [comment]
I have set:

```
    [markup.goldmark.renderer]
      unsafe = false
```

There are still XSS attacks that can be conducted in the search box

**Comment by colinwilson** [comment]
Yep.

Your initial assessment was correct, it's an issue.

Will fix this ASAP

**Comment by colinwilson** [comment]
Fixed by e4dbd49

P.S. The `unsafe = true` setting under `[markup.goldmark.renderer]` is necessary for a lot of shortcodes and other JavaScript to function properly. See the [Hugo documentation](https://gohugo.io/getting-started/configuration-markup/#goldmark) for more info:
> By default, Goldmark does not render raw HTML and potentially dangerous links. If you have lots of inline HTML and/or JavaScript, you may need to turn this on.

**Comment by PKUcoldkeyboard** [comment]
> Fixed by [e4dbd49](https://github.com/colinwilson/lotusdocs/commit/e4dbd493b1d27c97daf59f81210690d80ce6412b)
> 
> P.S. The `unsafe = true` setting under `[markup.goldmark.renderer]` is necessary for a lot of shortcodes and other JavaScript to function properly. See the [Hugo documentation](https://gohugo.io/getting-started/configuration-markup/#goldmark) for more info:
> 
> > By default, Goldmark does not render raw HTML and potentially dangerous links. If you have lots of inline HTML and/or JavaScript, you may need to turn this on.

Alright, I'll make sure to set `unsafe = true` to enable these awesome features. Thanks!

**Comment by colinwilson** [comment]
You're welcome. Thanks for spotting the issue. 👍
